### PR TITLE
fix(cli): inherit collection variables in folders without own variables

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Closes FE-1105

Collection runners weren't inheriting variables from parent collections into child folders when those folders didn't define their own variables. For example:

Collection (defines: `baseUrl=https://api.example.com`)
  └── Folder A (no variables defined)
      └── Request 1 (references `<<baseUrl>>`) → Can't access `baseUrl`

While this worked fine:

Collection (defines: `baseUrl=https://api.example.com`)
  └── Folder B (defines: `apiKey=secret`)
      └── Request 2 (references `<<baseUrl>>`) → Can access both `baseUrl` and `apiKey`

The issue was in the inheritance condition, where it checked whether the folder already had variables before merging parent variables, so empty folders never received anything.

### What's changed

- Fixed inheritance logic in `collections.ts` to check parent collection variables instead of folder variables.
- Initialise an empty `variables` array for folders that don't have one.
- Maintain correct precedence so folder variables override collection variables with the same key.
- Added comprehensive test coverage for empty folders, nested inheritance, and precedence scenarios.
- Removed unnecessary optional chaining checks.

### Notes to reviewers

The fix itself is straightforward - changed the condition from `if (updatedFolder.variables?.length)` to `if (collection.variables?.length`)` and added array initialisation. Most of the diff is comprehensive test coverage to prevent regression.

The enhanced test collection now covers:

- Folders without variables inheriting from the parent (the bug).
- Nested folders inheriting through the parent chain.
- Folder variables take precedence over collection variables.
- Request variables take precedence over collection variables.
- Root-level requests accessing collection variables.
- Variable precedence remains: `Request` > `Folder` > `Collection` > `Environment`.

---
## Summary by cubic
Fixes CLI collection variable inheritance so folders without variables inherit parent collection variables. Addresses Linear FE-1105; requests in empty folders can now access collection vars while keeping precedence intact.

- **Bug Fixes**
  - Merge collection variables into all folders, even when empty.
  - Preserve precedence: Request > Folder > Collection > Environment.
  - Add e2e tests and updated fixtures for empty/nested folders, overrides, and root-level requests.

<sup>Written for commit 2328afefca7c1501b77e19ebe8b6b728899ab1da. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/hoppscotch-cli/src/utils/collections.ts [hunk 1]
```diff
@@ -167,9 +167,9 @@ const processCollection = async (
       updatedFolder.headers.push(...filteredHeaders);
     }
 
-    if (updatedFolder.variables?.length) {
-      // Filter out variable entries present in the parent collection under the same name
-      // This ensures the folder variables take precedence over the collection variables
+    // Inherit collection variables into folder, with folder variables taking precedence
+    if (collection.variables?.length) {
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-cli/src/utils/collections.ts]: Collections are migrated to the latest version before this phase, and the `variables` property is guaranteed to exist, making the defensive check premature.
